libs/tasks/feature/src/lib/tasks-feature/tasks-feature.component.ts

```diff
@@ -1,30 +1,66 @@
-import { Component, OnInit } from '@angular/core';
-import { Task, TasksFacade } from '@todo-workspace/tasks/data-access';
-import { Observable } from 'rxjs';
+import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import {
+  PaginatorState,
+  Task,
+  TasksFacade,
+} from '@todo-workspace/tasks/data-access';
+import { Observable, Subscription } from 'rxjs';
 import { MatDialog } from '@angular/material/dialog';
 import { TaskFormDialogComponent } from '@todo-workspace/tasks/ui-task-form-dialog';
+import { SnackbarService } from '../../../../data-access/src/lib/services/snackbar.service';
 
 @Component({
   selector: 'todo-workspace-tasks-feature',
   templateUrl: './tasks-feature.component.html',
   styleUrls: ['./tasks-feature.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class TasksFeatureComponent implements OnInit {
   tasks$: Observable<Task[]>;
-  constructor(private tasksFacade: TasksFacade, private matDialog: MatDialog) {}
+  todoPage$: Observable<PaginatorState>;
+  donePage$: Observable<PaginatorState>;
+  loading$: Observable<boolean>;
+  subscription: Subscription;
+
+  constructor(
+    public tasksFacade: TasksFacade,
+    private matDialog: MatDialog,
+    private snackbar: SnackbarService
+  ) {}
 
   ngOnInit(): void {
     this.tasksFacade.init();
     this.tasks$ = this.tasksFacade.tasks$;
+    this.todoPage$ = this.tasksFacade.todoPage$;
+    this.donePage$ = this.tasksFacade.donePage$;
+    this.loading$ = this.tasksFacade.loading$;
+
+    this.subscription = this.tasksFacade.error$.subscribe((value) => {
+      this.snackbar.open(value);
+    });
   }
 
   openCreateDialog() {
+    this.openDialog('Add new task:', this.tasksFacade.create);
+  }
+
+  openEditDialog(task: Task) {
```

Comment:
do you need two functions to `openDialog`?

Reply:
no, but openEditDialog is called in two places, and then I should pass the same parameters two times, so I though that this more specific functions could be useful